### PR TITLE
Travis support

### PR DESCRIPTION
**Pull request by oferrigni**
Travis will install postgres + contrib, build and execute tests.
Docs suggest waiting for a passing build in travis for PRs.

### CODE REVIEW

**Comment by oferrigni** [comment]
Not trying to be opinionated. Didn't take long to do. Will need to setup at https://travis-ci.org/repositories to enable. Build passed in my fork @ https://travis-ci.org/oferrigni/epgsql-1 . README.md should point at https://travis-ci.org/epgsql/epgsql . Initial setup takes a bit, but runs pretty quick after that.

**Comment by oferrigni** [comment]
Just saw that this might answer https://github.com/epgsql/epgsql/issues/50

**Comment by seriyps** [comment]
Maybe add more OTP releases?

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  - R16B03-1
+before_script:
+  - sudo apt-get update -qq
+  - sudo apt-get install postgresql-9.3-postgis-2.1 postgresql-contrib
+addons:
+  postgresql: "9.3"
+script:
+  - sudo chmod 777 /var/run/postgresql/
+  - ./setup_test_db.sh
+  - nohup ./start_test_db.sh &
```
Comment (seriyps) [inline .travis.yml]: is `nohup` really required here?

Comment (oferrigni) [inline .travis.yml]: i think start_test_db.sh blocks

**Comment by oferrigni** [comment]
Did you have a list of otp versions you'd like? R16B03-1 is the minimum version and I didn't see any feedback that greater versions were being targeted on a roadmap.

File: .travis.yml (hunk 1 above)
Comment (seriyps) [inline .travis.yml]: yes, `start_test_db.sh` blocks, but I think just `&` may be enough. Anyway, this is not a principal question.

**Comment by seriyps** [comment]
As a rule of thumb, every library tries to be compatible with last 3 major Erlang releases. For now it is R15, R16, 17. Not sure about minor releases, but why not?
Full releases list can be found there http://www.erlang.org/download.html

**Comment by oferrigni** [comment]
epgsql does not compile in R15, I don't think.

**Comment by seriyps** [comment]
Oh, yeah. Looks like it doesn't compile on R15 because `query` was reserved word before R16 (https://github.com/erlang/otp/commit/0dc3a29744bed0b7f483ad72e19773dc0982ea50). This word is used in typespec, so, compilation failed with

```
Compiling src/epgsql.erl failed:
src/epgsql.erl:30: syntax error before: '('
src/epgsql.erl:122: syntax error before: 'query'
src/epgsql.erl:140: syntax error before: 'query'
src/epgsql.erl:158: syntax error before: 'query'
ERROR: compile failed while processing /home/seriy/workspace/epgsql: rebar_abort
```
When I renamed `-type query()` to smth else, it compiles with no problems. I think this should be reported separately and fixed. @davidw what do you think?

**Comment by davidw** [comment]
Yeah, that sounds like a separate issue that's worth fixing.

`start_test_db.sh` does block - it's kind of nice to be able to look at the queries going by.

**Comment by oferrigni** [comment]
Per PR comments, I have removed the nohup (which now intermingles queries with eunit output giving some context to what is happening behind the scenes) and I added 17.4 as the most recent published release. 

Passing build @ https://travis-ci.org/oferrigni/epgsql-1/builds/50362253

**Comment by oferrigni** [comment]
Are there other concerns or requested changes with this PR? I am worried I may have misunderstood some of the feedback. @seriyps @davidw

**Comment by seriyps** [comment]
I think it's ok now. More erlang releases can be added when `query` issue will be fixed later.

**Comment by davidw** [comment]
I merged it.... is there anything else we need to do to make it work?

**Comment by oferrigni** [comment]
I think you need to go to https://travis-ci.org/epgsql/epgsql and enable the hook. Requires your authorizing travis to access the repo.

**Comment by davidw** [comment]
Ok, I went and clicked some checkboxes on the Travis site.  We'll see if things start building...

**Comment by seriyps** [comment]
There is also docs http://docs.travis-ci.com/user/getting-started/#Step-two%3A-Activate-GitHub-Webhook

**Comment by davidw** [comment]
I can't tell if this is a problem with the Travis setup or just Travis catching a problem: https://github.com/epgsql/epgsql/pull/58

**Comment by seriyps** [comment]
copy-paste my investigation here too:
> looks like R15B03-1 isn't available on travis and this is known problem: https://github.com/erlydtl/erlydtl/blob/master/.travis.yml#L11
> I think we should replace R15B03-1 withR15B03 on travis.yml

**Comment by davidw** [comment]
Yep, looks good; done.

**Pull request merged**